lib/datasets/report_manager.rb

```diff
@@ -0,0 +1,72 @@
+require "datasets/report_summary"
+require "datasets/report"
+require "pathname"
+require "yaml"
+
+module Datasets
+
+  # A class that manages writing report_test.  Generally the primary
+  # avenue used to write report.  Contains functionality to discover
+  # the period of the most recent report and write a new one, as a sort
+  # of last_run_time manager.
+  class ReportManager
+
+    TIME_FORMAT = "%Y%m%d%H%M%S"
+
+    def initialize(parent_dir, fs)
+      @parent_dir = Pathname.new parent_dir
+      @fs = fs
+    end
+
+    # The period of the last completed report
+    # @return [Range<Time>]
+    def last_range
+      last_summary.time_range
+    end
+
+    # Finds the most recent report in the parent directory, then
+    # yields a new time range from the time the last report ended
+    # until now.  It then creates and saves a new report with the
+    # result of the block.
+    # @yieldparam new_range [Range<Time>] The period from when the last
+    #   report started until now.
+    # @yieldreturn [Array<Array<Volume>>] The block should return the
+    #   array of volumes that were saved and the array of volumes that
+    #   were removed.
+    # @return [Report]
+    def build_next_report(&block)
```

Comment:
The use of this is 
```
ReportManager.new('f/foo/bar', my_filesys).build_next_report do 
  # Something that ruturns [saved_volumes, deleted_volumes]
end
```
That's going to get called from the scheduler?  Will each scheduler being using it's own subdirectory?